include/amici/model.h

```diff
@@ -1737,13 +1750,19 @@ namespace amici {
         std::vector<int> plist_;
 
         /** state initialisation (size nx) */
-        std::vector<double>x0data;
+        std::vector<double> x0data;
 
         /** sensitivity initialisation (size nx * nplist, ordering = ?) */
-        std::vector<realtype>sx0data;
+        std::vector<realtype> sx0data;
 
         /** timepoints (size nt) */
-        std::vector<realtype>ts;
+        std::vector<realtype> ts;
+        
+        /** vector of flags indicating whether state variables are to be assumed to be positive */
+        std::vector<bool> qpositivex;
```

Comment:
Is there a specific reason for having a bool array instead of an index array? The latter would avoid iterating over nx elements in case there are no non-negativity constraints (and safe space).

Reply:
To copy `x`, we need to iterate over all elements anyways. Having an index array would require two loops. I think a cleaner solution would be to check once if any `qpositivex` is true and if not make computeX_pos not copy anything.